Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py

```diff
@@ -311,14 +318,24 @@ def is_checkout(cls, path):
         return run([cls.executable(), 'rev-parse', '--show-toplevel'], cwd=path, capture_output=True).returncode == 0
 
     @decorators.hybridmethod
-    def config(context):
+    @decorators.Memoize()
+    def config(context, location=None):
         args = [context.executable(), 'config', '-l']
         kwargs = dict(capture_output=True, encoding='utf-8')
+        if location and location not in context.CONFIG_LOCATIONS:
+            raise TypeError("'{}' is not a valid git config location".format(location))
 
-        if isinstance(context, type):
+        if isinstance(context, type) or location == 'global':
+            if location in ['repository', 'project']:
+                raise TypeError("Cannot find '{}' git config without repository".format(location))
             args += ['--global']
         else:
             kwargs['cwd'] = context.root_path
+            if location == 'project':
+                # Without a project config, use the library defaults
+                if not os.path.isfile(os.path.join(context.root_path, context.PROJECT_CONFIG_PATH)):
+                    return {key: values[0] for key, values in context.PROJECT_CONFIG_OPTIONS.items()}
+                args += ['--file', context.PROJECT_CONFIG_PATH]
```

Comment:
This is a good idea, which I'm going to do, but I don't know that it helps us get rid of this code, because we don't have any guarantees that folks have run `setup` in this function, and given that it's actually used in `setup`, we can be very certain they have not.